### ISSUE TITLE
as_track for data frames with x_, y_, and t_ variables

### ISSUE THREAD

**Issue by bniebuhr** [jmsigner/amt#44]
Hello,

I have been using the `amt` package to perform operations with individuals separately, such as re-sample, calculate movement variables by burst annotating data etc. Generally I have a `track_xyt` object and, following the examples from the `amt` paper, I use `tidyr::nest` before calculating the variables for each individual. Sometimes this involves grouping variables by burst or some unit of time (day, week, month) to calculate extra movement variables (e.g., summaries of mean, maximum or cumulative distance traveled during the time frame). However, when doing these operations, sometimes the classes `track_xy` and `track_xyt` are lost in the unnested output, and it keeps only the classes `data.frame`, `tbl_df`, and `tbl`. I do not have a reproducible example here in hands, but I can come back here as soon as this happens again.

In any case, with that on mind (and since this has happened several times to me already), I was thinking about an `as_track` method for `data.frame`s that used to be `track_xy` or `track_xyt` object but "lost" these classes. In this case, they already have the variables called `x_` and `y_` (and maybe `t_`), but these classes must be enforced into them again.

Looking at the other `as_track` methods, it would look something like (not sure if the `function_name.data.frame` works because of the dots):

```r
#' @export
#' @rdname as_track
as_track.data.frame <- function(x, ...) {
  cols <- colnames(x)

  if("x_" %in% cols & "y_" %in% cols & !("t_" %in% cols)) {
    make_track(x = x$x_, y = y_, ...)
  } else {
    if("x_" %in% cols & "y_" %in% cols & "t_" %in% cols) {
      make_track(x = x$x_, y = x$y_, t = x$t_, ...)
    }
  }
  
}
```

I think this might be useful in these cases, so I keep here as suggestion. I can also implement it and make a PR, but it would be nice to have a double check by someone else. And more than that, to discuss if this would really be worth implementing.

**Comment by jmsigner** [comment]
This seems to be very useful. Feel free to add a PR. I will do soon a bunch of updates to `amt` and would be happy to include this.

**Comment by jmsigner** [comment]
This is now implemented in the development version (9335b0964405b653e4815db990c56321cdcaf043). Thanks for suggesting this. I added you as a contributor.

**Comment by bniebuhr** [comment]
Oh, that's nice! I'll test it asap.
